cmd/cluster-agent/webhook/server.go

```diff
@@ -0,0 +1,145 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+/*
+Package api implements the agent IPC api. Using HTTP
+calls, it's possible to communicate with the agent,
+sending commands and receiving infos.
+*/
+package webhook
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"github.com/DataDog/datadog-agent/pkg/api/security"
+	stdLog "log"
+	"net"
+	"net/http"
+	"strings"
+
+	"github.com/gorilla/mux"
+
+	"github.com/DataDog/datadog-agent/cmd/cluster-agent/webhook/agent"
+	"github.com/DataDog/datadog-agent/pkg/api/util"
+	"github.com/DataDog/datadog-agent/pkg/clusteragent"
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+var (
+	listener net.Listener
+)
+
+// StartServer creates the router and starts the HTTP server
+func StartServer(sc clusteragent.ServerContext) error {
+	// create the root HTTP router
+	r := mux.NewRouter()
+
+	// IPC REST API server
+	agent.SetupHandlers(r, sc)
+
+	// get the transport we're going to use under HTTP
+	var err error
+	listener, err = getListener()
+	if err != nil {
+		// we use the listener to handle commands for the agent, there's
+		// no way we can recover from this error
+		return fmt.Errorf("Unable to create the api server: %v", err)
+	}
+
+	var TLSCert tls.Certificate
```

Comment:
```suggestion
	var cert tls.Certificate
```
Capitalisation is confusing in Go because it means "exporter identifier". What you did is ok because the convention in Go is to capitalise acronyms, but here it's a local variable which is lowercase so there is an exception. You can also use `tlsCert` if you like that more...

Reply:
Addressed 👍